### ISSUE TITLE
Mark is not visibile after expand-region

### ISSUE THREAD

**Issue by kandiu** [magnars/expand-region.el#229]
Hello, I have this issue with expand region.

1. Use expand region to select some text ( the mark is visible)
2. Do something with the selected text
3. Use emacs default Mark set (C-space) to activate transient mark mode. 

Now, the selection is not visibile. I mean, the selection is not visibile, but the text is getting selected. Because I can kill and yank, it works. I just don't have the visual feedback of what is selected. This happens only if I first use expand-region and the transient mark mode.

**Comment by yunxing** [comment]
Getting the same issue.

**Comment by billturner** [comment]
Not sure if it's the same problem as #220, although likely related. A fix was committed for that bug, but the issue still seems to persist for me.

(I'm very new to emacs, and not sure what could be the problem, but happy to help if I can.)

**Comment by kandiu** [comment]
I agree, the problem seems the one described in #220. I cannot fix either.

**Comment by wyuenho** [comment]
I have the same problem too. I still had to turn off `shift-select-mode` to deal with this issue even after #225 was merged. I was wondering if you guys have multiple-cursors turned on?  Multiple cursor's `rectangular-region-mode` seems to mess with `expand-region` too.

**Comment by Bad-ptr** [comment]
I can reproduce this if I type something to replace or delete a region selected using expand-region.

Steps to reproduce:
1) place the point on a symbol/word
2) invoke er/expand-region
3) see the current symbol/word became selected
4) type `3`
5) see the current symbol/word replaced by `3`
6) `M-: transient-mark-mode` see `(only . t)`
7) Press `C-SPC`, move point, see no highlight on a region.
8) Press `C-SPC` again, move point and see that the region is highlighted.

**Comment by mallt** [comment]
In order to support extending the region with `shift-select-mode` after using `expand-region`, `expand-region` uses the pair notation `(only. OLDVAL)` of the `transient-mark-mode` variable. When "leaving" `expand-region`, the value of `transient-mark-mode` should be reset to the original value before starting to expand, but currently this only happens when using `C-g` or `0`. The case of leaving `expand-region` by inserting text is f.ex. not handled, and therefore highlighting the region afterwards with `C-SPC` is not working because the `transient-mark-mode` variable is still in the pair notation `(only. OLDVAL)` for `shift-select-mode`.

@magnars, do you see a possible way of detecting the occurence of "leaving" `expand-region`, so the `transient-mark-mode` variable can be reset to its original state?

**Comment by samclearman** [comment]
Another way of "leaving expand-region", one which I imagine is pretty common, is doing a replace-string.  This is bad for users with shift-select enabled, much worse than https://github.com/magnars/expand-region.el/issues/190 in my opinion.

If rolling back https://github.com/magnars/expand-region.el/commit/acef6cc130dfc8f9b06677330cb87e4119ceb88b will fix this issue, my vote would be to roll it back.

**Comment by magnars** [comment]
> If rolling back acef6cc will fix this issue, my vote would be to roll it back.

Agreed. This changed was causing too many issues. Reverted.
